### PR TITLE
Add an article role and allow quick navigation to articles

### LINKED ISSUE #9227: Support quick navigation by article in browse mode
### Is your feature request related to a problem? Please describe.

I propose that NVDA implement a quick navigation key for browse mode that allows to skip back and forward by articles. So far, this is done via the Enhanced ARIA add-on, which simply expands the landmark navigation to include articles. However, with many articles, the rest of the landmarks get drowned out easily, so it makes sense to use a separate keystroke specifically for articles.

### Describe the solution you'd like

* Introduce a new keystroke in quick nav scripts to jump back and forward between article elements on a page. Alternatively, repurpose the keystroke previously used to navigate among Flash objects. Flash is dead, so it makes sense to repurpose this keystroke. Mozilla announced that Flash will be gone in Firefox 69, and object navigation, with the last of the NPAPI plugins gone, is then obsolete.
* When navigating with the virtual cursor, announce when the user enters a new article similar to what is being done for other landmarks.

### Describe alternatives you've considered

Using the Enhanced ARIA add-on, which works on and off depending on the new NVDA add-on compatibility feature, making this unreliable, but the need for this kind of navigation is definitely there. Also, see above, using the same landmark navigation also for articles drowns other landmarks.

### Additional context

Many websites like the new Twitter, AKA [Mobile Twitter](https://mobile.twitter.com), or [Threema Web](https://web.threema.ch), or the [Mastodon](https://mastodon.social) or [Pinafore](https://pinafore.social) Mastodon clients use articles to separate one message or tweet from one another. In addition, the [Feed design pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#feed) in WAI-ARIA 1.1 encourages to group entities by elements with the article role to separate them and make them navigable.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #9227 

### Summary of the issue:
It has been a long standing request to allow quick navigation from/to articles. Also, articles should be able to mentioned as such.

### Description of how this pull request fixes the issue:
1. Added a new article role, which is set to articles in virtual buffers and on NVDAObjects that are articles.
2. Added the possibility for quick navigation scripts to be unbound by default.
3. Added a quicknav script for articles, working in MSHTML and IA2 based browsers.

### Testing performed:
Tested the following in Firefox, Chrome and IE:

1. Articles are mentioned at such when using the <article> tag or the article aria role when reporting of articles is enabled
2. Quicknav works after assigning keys to the scripts.

### Known issues with pull request:
The implementation for EdgeHTML is limited in that it works for nodes that have been marked up with the aria role of article. Articles marked up as such with the `<article>` tag aren't recognized, because they have the grouping role and the only way to detect whether it's an article is by means of the `LocalizedControlType`. Bah!

### Change log entry:
* New features
    + In Internet Explorer, Google Chrome and Mozilla Firefox, You can now navigate by article using quick navigation scripts. These scripts are unbound by default and can be assigned in the Input Gestures dialog when the dialog is opened from a browse mode document. (#9227)
* Changes for developers
    + The aria.htmlNodeNameToAriaLandmarkRoles dictionary has been renamed to aria.htmlNodeNameToAriaRoles. It now also contains roles that aren't landmarks.

### CODE REVIEW

**Comment by leonardder** [comment]
Just did a quick test with the UIA implementation of Edge Chromium. There, it looks like the aria role of article will be properly exposed, even when using the `<aticle>` tag. Therefore, I think that we should ignore the limitation in Edge for now.

File: source/browseMode.py [hunk 1]
```diff
@@ -416,10 +417,20 @@ def _quickNavScript(self,gesture, itemType, direction, errorMessage, readUnit):
...
+			itemType: str,
+			key: Optional[str],
+			nextDoc: str,
+			nextError: str,
+			prevDoc: str,
+			prevError: str,
+			readUnit: Optional[str] = None
+	):
 		"""Adds a script for the given quick nav item.
-		@param itemType: The type of item, I.E. "heading" "Link" ...
```
Comment (michaelDCurran) [inline source/browseMode.py]: Why was this line removed?

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

File: source/browseMode.py (hunk 1 above)
Comment (leonardder) [reply source/browseMode.py]: Ugh, that must have been an accident. Restoring now.

**Comment by leonardder** [comment]
In the current implementation, the "out of article" sentence is also spoken when leaving articles. For landmarks, this isn't the case. While landmarks and articles aren't the same, I feel they should behave equivalent in this case. Thus, either they should announce exit, or they shouldn't. See also #10420

**Review by michaelDCurran — APPROVED** [review]
